v0x04/match_fields.py

```diff
@@ -37,12 +37,15 @@ def as_of_tlv(self):
         """Return a pyof OXM TLV instance."""
         try:
             value = int(self.value)
+            value = value | VlanId.OFPVID_PRESENT
+            if value == 0:
```

Comment:
@Alopalao, this conditional doesn't look coherent, could you double check your implementation? `value | VlanId.OFPVID_PRESENT` where `VlanId.OFPVID_PRESENT = 0x1000` will never be zero, so this conditional is unreachable. 

Also, I'm not sure if how historically the mask is being set here if it's correct, I'd suggest to first also compare with other switches like OvS, in addition to reviewing the text of the specification. I've mapped this issue https://github.com/kytos-ng/of_core/issues/99 here to explore these points, let me know if you'd like to explore it, I could also help to exchange some ideas if you need. 

Thanks.

Reply:
You are right, my bad.

Reply:
The updated code looks good. Let me close this thread.